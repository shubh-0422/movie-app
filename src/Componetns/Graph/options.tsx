import { Options } from "highcharts";
export const options: Options = {
    chart: {
      zoomType: "xy",
      backgroundColor: "#303030",
      height: 220,
    },
    credits: {
      enabled: false,
    },
    title: undefined,
    subtitle: undefined,
    yAxis: [
      {
        labels: {
          format: "{value}",
          style: {
            color: "#aaaaaa",
          },
        },
        title: {
          text: null,
          style: {
            color: "#aaaaaa",
          },
        },
        opposite: true,
      },
      {
        title: {
          text: null,
          style: {
            color: "#aaaaaa",
          },
        },
        labels: {
          format: "{value}thm",
          style: {
            color: "#aaaaaa",
          },
        },
      },
      {
        gridLineWidth: 1,
        gridLineColor: "#777777",
        title: {
          text: null,
          style: {
            color: "#aaaaaa",
          },
        },
        labels: {
          formatter: function (object) {
            const formattedValue = parseInt(object.value.toString()) / 1000;
            return formattedValue === 0
              ? "$" + formattedValue
              : "$" + formattedValue + "K";
          },
          style: {
            color: "#aaaaaa",
          },
        },
        opposite: true,
      },
    ],
    tooltip: {
      shared: true,
    },
    legend: {
      layout: "horizontal",
      align: "center",
      x: 0,
      verticalAlign: "bottom",
      y: 0,
      floating: false,
      itemStyle: { color: "#aaaaaa" },
      itemHoverStyle: { color: "#dddddd" },
    },
    responsive: {
      rules: [
        {
          condition: {
            maxWidth: 500,
          },
          chartOptions: {
            legend: {
              floating: false,
              layout: "horizontal",
              align: "center",
              verticalAlign: "bottom",
              x: 0,
              y: 0,
            },
            yAxis: [
              {
                labels: {
                  align: "right",
                  x: 0,
                  y: -6,
                },
                showLastLabel: false,
              },
              {
                labels: {
                  align: "left",
                  x: 0,
                  y: -6,
                },
                showLastLabel: false,
              },
              {
                visible: false,
              },
            ],
          },
        },
      ],
    },
  };