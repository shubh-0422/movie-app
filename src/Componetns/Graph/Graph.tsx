import moment from "moment";
import { options } from "./options";
import {data} from './../../Data/Data';
import Highcharts from "highcharts";
import HighchartsReact from "highcharts-react-official";
import './Graph.scss';


const Graph = () => {
    const monthwiseAggregations =data.data.regions[0].facilities[0].monthwiseAggregations;

  options.xAxis = [
    {
      categories: monthwiseAggregations.map((item) =>
        moment(item.monthStartDate).format("MMM")
      ),
      crosshair: true,
    },
  ];

  options.series = [
    {
      name: "thm Baseline",
      type: "column",
      yAxis: 1,
      color: "#606062",
      borderWidth: 0,
      data: monthwiseAggregations.map((item) => item.baselineConsumption),
      tooltip: {
        valueSuffix: "thm",
      },
    },
    {
      name: "thm Actual",
      type: "column",
      yAxis: 1,
      color: "#1371b4",
      borderWidth: 0,
      data: monthwiseAggregations.map((item) => item.actualConsumption),
      tooltip: {
        valueSuffix: "thm",
      },
    },
    {
      name: "Baseline Cost",
      type: "line",
      dashStyle: "ShortDash",
      yAxis: 2,
      color: "#a1762b",
      data: monthwiseAggregations.map((item) => item.baselineCost),
      tooltip: {
        valuePrefix: "$",
      },
    },
    {
      name: "Actual Cost",
      type: "line",
      yAxis: 2,
      color: "#e5a743",
      data: monthwiseAggregations.map((item) => item.actualCost),
      tooltip: {
        valueSuffix: "$",
      },
    },
  ];

  return <HighchartsReact highcharts={Highcharts} options={options} />;
};

export default Graph;