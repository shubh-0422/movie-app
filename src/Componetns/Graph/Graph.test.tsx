import { mount, shallow, ReactWrapper,configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Graph from "./Graph";

configure({adapter : new Adapter() });

describe("Graph Component", ()=> {
    it("should render without crash", () => {
        const Component = <Graph />;
        const wrapper = mount(Component);
        expect(wrapper).toBeTruthy();
      });

      it('should render correctly with no props', () => {
        const wrapper = shallow(<Graph />);
        expect(wrapper).toMatchSnapshot();
      });
})