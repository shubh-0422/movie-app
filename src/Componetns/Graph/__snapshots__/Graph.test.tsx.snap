// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Component should render correctly with no props 1`] = `ShallowWrapper {}`;

exports[`Graph Component should render correctly with no props 1`] = `ShallowWrapper {}`;
